Lib/test/test_time.py

```diff
@@ -160,6 +173,8 @@ def test_sleep(self):
         self.assertRaises(ValueError, time.sleep, -2)
         self.assertRaises(ValueError, time.sleep, -1)
         time.sleep(1.2)
+        time.sleep(IndexLike(0))
+        time.sleep(FloatLike(0))
```

Comment:
I don't entirely understand these tests (they don't seem to be asserting anything other than "does not raise an exception"), but presumably we also want to do something like `time.sleep(Decimal(0))` and/or the case mentioned in the issue, `time.sleep(Decimal('0.99'))` (though it would be much better if we could make an assertion about the duration of the sleep  in the `Decimal('0.99')` case).

Reply:
> they don't seem to be asserting anything other than "does not raise an exception"

Indeed. Testing that it doesn't raise an exception is the whole point of these tests. What's wrong with that?

Reply:
> Testing that it doesn't raise an exception is the whole point of these tests. What's wrong with that?

There's nothing especially wrong with it if it's the best you can do, but usually you get the "doesn't raise an exception" test for free when you test that it does the right thing. Admittedly `time.sleep` is quite hard to test, so maybe this is the best we can do. I usually just worry that if I'm not asserting anything about the result of a test function I am missing an opportunity for a stronger test.